src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/service/CourtService.kt

```diff
@@ -64,10 +65,17 @@ class CourtService(
     }
 
     with(courtInsertRecord) {
-      val court =
-        Court(courtId, courtName, courtDescription, courtTypeRepository.findById(courtType).orElseThrow(), active)
+      val court = Court(courtId, courtName, courtDescription, courtTypeRepository.findById(courtType).orElseThrow(), active)
       courtRepository.save(court)
-      return CourtDto(court)
+
+      buildings?.forEach {
```

Comment:
I was wondering if we could get hibernate to do all this for us - by modelling the relationships in the JPA model?

Reply:
yes it can and normally I would do this - only I chickened out as Kotlin was confusing me at first.  I'd like to revisit it.  It does have a cascade on - just need the "add" methods to setup backlinks etc